packages/common/src/types/semanticLayer.ts

```diff
@@ -162,32 +233,79 @@ export type SemanticLayerFilterBase = {
 };
 
 export type SemanticLayerStringFilter = SemanticLayerFilterBase & {
-    operator: SemanticLayerStringFilterOperator;
+    operator: SemanticLayerFilterBaseOperator;
+    values: string[];
+};
+
+export type SemanticLayerExactTimeFilter = SemanticLayerFilterBase & {
+    operator: SemanticLayerFilterBaseOperator;
     values: string[];
 };
 
-// TODO: right now we only support string filters, this type should be a union of all filter types
-type SemanticLayerFilterTypes = SemanticLayerStringFilter;
+export type SemanticLayerRelativeTimeFilter = SemanticLayerFilterBase & {
+    operator: SemanticLayerFilterBaseOperator;
+    values: undefined;
+    relativeTime: SemanticLayerFilterRelativeTimeValue;
```

Comment:
basically, this I don't think belongs in this type but should be a UI only thing. eventually - we gotta generate a value for it.

I think it eventually should end up in the values. for example. for `SemanticLayerRelativeTimeFilter` we can have an "advanced" values type which can be an object.

{ time: number with millisecond precision, relevant: boolean (default false) }
so one week before would be `{ time: -604800000 (1 week in milliseconds with minus): relevant: true }`

Reply:
or we can come up with even stronger types. let me know if you wanna chat about this. we can make different types for different field types + operator combinations. happy to design this with you!